### DISCUSSION TITLE
Artisan issues with Donyi PKF Roaster

### DISCUSSION THREAD

**Discussion by EvergreenCoffee** [artisan-roaster-scope/artisan#1403] [Q&A]
Hello!

Im not a computer wiz by any stretch but I am struggling to connect artisan with my roaster, according to Dongyi its easily doable and of course is proving to be not so much. I have the prescribed drivers downloaded. My temp reading fluctuate however are much higher than actual readings which Im assuming is an indication of an issue. 

I saw something regarding someone using a widget but there weren't any details on this. Has anyone had success connecting one of these that can help me out?

Thanks in advance and apologies if this question has been posed, I didn't see anything when I looked!

**Comment by poundy** [comment]
Did you check out the fundamentals at https://artisan-scope.org/machines/dongyi/ ? 
Did you set up the machine using the inbuilt machine selection process?
When you say "my temp reading fluctuate" do you mean that Artisan shows you temperatures? If so that means Artisan is connected to the roaster correctly ! Please clarify.

If temperatures are not showing "correct" then the most obvious problem is a C vs F setting somewhere. Do some stable state measurements and understand what the machines displays report and what Artisan is reading. Then you'll need to look at the machine settings to correct this. Even if the display temperature is F, I understand that most PIDs will report in C by default, so you may just need to adjust your setting to C on the machine settings page in Config -> Port -> Modbus, changing mode from F to C.
